Yona/Yona/Challenges/TimeBucketChallenges.swift

```diff
@@ -260,6 +268,13 @@ class TimeBucketChallenges: UIViewController,UIScrollViewDelegate {
     
     // MARK: - Tapgesture category view tap events
     func categoryTapEvent(sender: UITapGestureRecognizer? = nil) {
+        if self.activityCategoriesArray.count > 0 {
+            self.addNewGoalButton.hidden = false
+        } else {
+            dispatch_async(dispatch_get_main_queue()) {
+                self.addNewGoalButton.hidden = true
+            }
+        }
```

Comment:
@chandanvarma009 no need for if else. can't you just give self.activityCategoriesArray.count > 0 to self.addNewGoalButton.hidden